packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -614,8 +615,13 @@ class _FocusState extends State<Focus> {
 
   void _handleAutofocus() {
     if (!_didAutofocus && widget.autofocus) {
-      FocusScope.of(context).autofocus(focusNode);
       _didAutofocus = true;
+      // Waiting for inactive nodes to be disposed.
+      SchedulerBinding.instance!.addPostFrameCallback((_) {
```

Comment:
I think ideally we should fix `FocusScopeNode.autofocus`.

Reply:
I tried it, but the inactive node may become active again.
 For example, by the global key or the node's attachment is rewritten, so we need to wait until finalizetree to confirm whether the pending node has a chance to regain focus, which is not very easy to do.